### DISCUSSION TITLE
active/standby failover using external UDP load balancer

### DISCUSSION THREAD

**Discussion by doka380** [strongswan/strongswan#1487] [General]
Hi, has anyone built (or at least was trying to build) active/standup failover for Strongswan using external UDP load balancer? The idea is to have two nodes - primary and backup, which are independent, just having the same configuration. The idea behind is the following:

- no load-balancing under normal circumstances. There is primary node which serves clients and hot backup node.
- If primary node fails (due to any reason), then all sessions will be re-established to backup node (no session/states persistence).
- When primary node will recover, all new sessions will be established to it, while existing sessions will remain on backup node until explicitly reestablished (to avoid unnecessary drop).

Any ideas whether this is possible? Any changes to scenario if smth is impossible to achieve? Any other issues? Any suggestions on UDP load-balancer for use (there are plenty of choices, starting from NGINX toward less known)?

I have read https://docs.strongswan.org/docs/5.9/features/highAvailability.html and, as far as I got, (a) this is experimental feature and (b) it require nodes to be inside a single LAN, while I'm going to use geographically different data centers which seems to be a stopper for bundled HA.

Will appreciate any feedback on that topic. Thank you.

**Comment by misch42** [comment]
Hi,

load balancing makes sense if you have more than two nodes and no single node is capable of dealing with the whole load. Or you have a loadbalancer anyway. But in a simple two-node setup there is no need for a loadbalancer.

Just set up a simple VRRP system with keepalived. Much easier and much more reliable.

Michael.

**Reply by doka380** [reply]
Hi Michael,

kind of professional deformation - in one of my project LBs heavily used for this purpose (historically) and I used to think like that :-) Sure, 'keepalived' approach is simpler.

Thank you.
